vpr/src/base/vpr_constraints_serializer.h

```diff
@@ -58,9 +58,34 @@ class VprConstraintsSerializer final : public uxsd::VprConstraintsBase<VprConstr
     virtual inline void set_add_atom_name_pattern(const char* name_pattern, void*& /*ctx*/) final {
         auto& atom_ctx = g_vpr_ctx.atom();
         std::string atom_name = name_pattern;
+
+        auto atom_name_regex = std::regex(atom_name);
+
+        std::vector<AtomBlockId> clear_atoms;
+        atoms_ = clear_atoms;
+
         atom_id_ = atom_ctx.nlist.find_block(name_pattern);
 
-        if (atom_id_ == AtomBlockId::INVALID()) {
+        if (atom_id_ != AtomBlockId::INVALID()) {
```

Comment:
comment here

Reply:
might have a specific name, might have a regex, this code could get slow if you have a lot of wildcards